### PR TITLE
fix: Handle unicode minus in number parsing

### LINKED ISSUE #6206: [Bug]: Unicode minus not parsed correctly
### Verified issue does not already exist?

- [x] I have searched and found no existing issue

### What happened?

When generating a CSV file for import from a XLS using Numbers on macOS, the file contained unicode minus instead of ascii minus causing the parsing to only end up with positive numbers. A quick search and replace fixed the issue, but suggest adding parsing for this character as well to make things just work. 
 
* Unicode minus U+2212 (“−”)
* ASCII hyphen-minus ("-")

### How can we reproduce the issue?

Import the following example CSV:

Date;Text;In/Out;Sum
2025-02-02;Systembolaget;−100,00;900.00
2025-02-02;Systembolaget;-100,00;800.00

Only the second entry is registered as negative. 

### Where are you hosting Actual?

Desktop App (Electron)

### What browsers are you seeing the problem on?

_No response_

### Operating System

Mac OSX

### PR DESCRIPTION
**Pull request by MatissJanis**
Fixes #6206.

When importing CSV files generated by Numbers on macOS, amounts may contain the Unicode minus sign (U+2212) instead of the ASCII hyphen-minus. This caused parsing to fail (resulting in positive numbers instead of negative).

This PR updates `looselyParseAmount`, `currencyToAmount`, and `stringToInteger` in `loot-core/src/shared/util.ts` to replace Unicode minus with ASCII minus before parsing.

Added unit tests to cover these cases.

---
<a href="https://cursor.com/background-agent?bcId=bc-9db3987c-5c4f-4548-b865-18ab0b320957"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/open-in-cursor-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/open-in-cursor-light.svg">Open in Cursor</picture></a>&nbsp;<a href="https://cursor.com/agents?id=bc-9db3987c-5c4f-4548-b865-18ab0b320957"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/open-in-web-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/open-in-web-light.svg">Open in Web</picture></a>

---

> [!NOTE]
> Normalize U+2212 to '-' in parsing utilities and add tests to ensure negative values are handled correctly.
> 
> - **Core utilities**:
>   - **Number parsing**: `currencyToAmount`, `stringToInteger`, and `looselyParseAmount` now normalize Unicode minus (`\u2212`) to ASCII `-`; parentheses + Unicode minus handled correctly.
> - **Tests**:
>   - Add unit tests covering Unicode minus across parsing scenarios.
> - **Release notes**:
>   - Add `upcoming-release-notes/6247.md` entry documenting the bugfix.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 2728303adf73159488eb984d2f94a849874c9b60. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Comment by youngcw** [comment]
Works in my testing

File: packages/loot-core/src/shared/util.ts [hunk 1]
```diff
@@ -518,6 +522,8 @@ export function integerToAmount(
 // number format, because the user could be importing from many
 // currencies. We extract out the numbers and just ignore separators.
 export function looselyParseAmount(amount: string) {
+  amount = amount.replace(/\u2212/g, '-');
```
[the remark this answers is not in the document]
Comment (MatissJanis) [reply packages/loot-core/src/shared/util.ts]: @cursor fix this please

**Review by youngcw — APPROVED** [review]
